transactions/src/main/java/org/jboss/as/txn/subsystem/TransactionSubsystemAdd.java

```diff
@@ -473,6 +474,14 @@ private void performRecoveryEnvBoottime(OperationContext context, ModelNode mode
                 .addService(TxnServices.JBOSS_TXN_EXTENDED_JBOSS_XA_TERMINATOR, extendedJBossXATerminatorService)
                 .setInitialMode(Mode.ACTIVE).install();
 
+        final JBossContextXATerminatorService contextXATerminatorService = new JBossContextXATerminatorService();
+        context.getServiceTarget()
+                .addService(TxnServices.JBOSS_TXN_CONTEXT_XA_TERMINATOR, contextXATerminatorService)
```

Comment:
Just an FYI here: dependency ordering is addressed by the Capabilities and Responsibilities functionality (https://docs.jboss.org/author/display/WFLY10/Working+with+WildFly+Capabilities) - at some point we need to expose all of our transaction subsystem services as capabilities (but not right now).